### ISSUE TITLE
Docker fails - Fixes included

### ISSUE THREAD

**Issue by iquaba** [google/timesketch#1174]
After `docker-compose up` the command fails with the error: `ModuleNotFoundError: No module named 'setuptools'`

**FIX 1:** I fixed this by adding `RUN apt-get -y install python3-setuptools` to the Dockerfile just before `RUN pip3 install timesketch`

At this point the container is running and the webserver is up, however I could not log in.

The next error I found was: `timesketch_1     | pkg_resources.DistributionNotFound: The 'idna<2.9,>=2.5' distribution was not found and is required by requests`

**FIX 2:** I fixed this by adding `RUN pip3 install 'idna<2.6,>=2.5'` to the Dockerfile before `RUN pip3 install timesketch`

Everything seems to be working now.

**Comment by Kamforka** [comment]
Experienced exactly the same issues, and more or less did the same patches, except that I installed them like this: `pip install timesketch idna==2.8`

I guess the `idna` issue might be due to the version mismatch of the `requests` library required by `plaso` and `timesketch`.

**Comment by iquaba** [comment]
Thanks @Kamforka - that worked for me as well but I think you meant `**pip3** install timesketch idna==2.8`

**Comment by Kamforka** [comment]
@iquaba Indeed the command is `pip3` in the docker build context!

**Comment by temach** [comment]
Issue with idna version was fixed in commit [15f9ab94](https://github.com/google/timesketch/commit/15f9ab9482af6128b336928be989473d29bedf56) by bumping requests library version (which uses  idna under the hood).

However two problems with docker-compose remain, see the error logs below.

Problem 1: no setuptools
```
Removing intermediate container 8b2e8d5eb9bd
 ---> d924bdabb275
Step 7/17 : RUN pip3 install timesketch
 ---> Running in f149b97f443d
Collecting timesketch
  Downloading https://files.pythonhosted.org/packages/ed/00/4fd5de4724b90719cf7777e19f5144bd6fa040333b382ced2d8ef4785d24/timesketch-20200507.tar.gz (8.9MB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'setuptools'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-qpnk56ax/timesketch/
ERROR: Service 'timesketch' failed to build: The command '/bin/sh -c pip3 install timesketch' returned a non-zero code: 1
```


Problem 2: no wheels package
```
Building wheels for collected packages: timesketch, PyYAML, SQLAlchemy, alembic, flask-bcrypt, flask-login, flask-script, neo4jrestclient, tabulate, toolz, pyrsistent, wrapt
  Running setup.py bdist_wheel for timesketch: started
  Running setup.py bdist_wheel for timesketch: finished with status 'error'
  Complete output from command /usr/bin/python3 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-oih4cb4i/timesketch/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpaj7kblimpip-wheel- --python-tag cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for timesketch
  Running setup.py clean for timesketch
  Running setup.py bdist_wheel for PyYAML: started
  Running setup.py bdist_wheel for PyYAML: finished with status 'error'
```

This can be fixed by installing python3-setuptools and python3-wheels packages. Without it all the wheels fail to build.

**Comment by binglot** [comment]
On the similar subject, Docker fails when building the "development_timesketch" container. Basically, the `requirements.txt` file states that `requests` needs to be in version 2.21.0, while `pymisp` requires at least 2.22.0. As a result, you'll get the below errors and you can access Timesketch via browser but it won't let you log in (it doesn't throw any errors at all, weirdly enough).

```
$ sudo docker logs -f $CONTAINER_ID

[...]
0.1->Flask->Flask-Testing>=0.6.2->-r /usr/local/src/timesketch/test_requirements.txt (line 1))
Building wheels for collected packages: Flask-Testing
  Running setup.py bdist_wheel for Flask-Testing: started
  Running setup.py bdist_wheel for Flask-Testing: finished with status 'done'
  Stored in directory: /root/.cache/pip/wheels/17/8d/ee/08cf29d904316ac2714cfe1dd2698ba36b00d8d6b2781310be
Successfully built Flask-Testing
Installing collected packages: Flask-Testing, mock, nose, pbr, soupsieve, beautifulsoup4, coverage, toml, lazy-object-proxy, typed-ast, astroid, isort, mccabe, pylint
Successfully installed Flask-Testing-0.8.0 astroid-2.4.1 beautifulsoup4-4.9.1 coverage-5.1 isort-4.3.21 lazy-object-proxy-1.4.3 mccabe-0.6.1 mock-4.0.2 nose-1.3.7 pbr-5.4.5 pylint-2.5.2 soupsieve-2.0.1 toml-0.10.1 typed-ast-1.4.1
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (requests 2.21.0 (/usr/lib/python3/dist-packages), Requirement.parse('requests<3.0.0,>=2.22.0'), {'pymisp'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/tsctl", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 3252, in <module>
    @_call_aside
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 3236, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 3265, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/local/lib/python3.6/dist-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (requests 2.21.0 (/usr/lib/python3/dist-packages), Requirement.parse('requests<3.0.0,>=2.22.0'), {'pymisp'})
Timesketch development server is ready!
```

**Comment by berggren** [comment]
Hi, this has been fixed in #1207 as part of a larger refactor and fixit for docker in TS. We now build base images automatically that is used.A

**Comment by berggren** [comment]
Please reopen if there is still issues.
